custom/git-app-deployer/main.go

```diff
@@ -0,0 +1,178 @@
+package main
+
+import (
+	"bytes"
+	"flag"
+	"os/exec"
+	"path/filepath"
+	"time"
+
+	"github.com/litmuschaos/test-tools/pkg/log"
+	"github.com/openebs/maya/pkg/util/retry"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/util/homedir"
+)
+
+func main() {
+
+	namespace := "sock-shop"
+
+	filePath, timeout := GetData()
+
+	config, err := GetKubeConfig()
+	if err != nil {
+		panic(err.Error())
+	}
+
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		panic(err.Error())
+	}
+
+	log.Info("[Status]: App-deployer is started...")
+	log.Info("[Status]: App-deployer type is: " + filePath)
+
+	if err := CreateNamespace(clientset, namespace); err != nil {
+		log.Fatalf("Failed to create the namespace due to, err: %v", err)
+	}
+	log.Info("[Status]: Namespace shock-shop is successfully created!")
+
+	if err := CreateShockShop(`/var/run/`+filePath, namespace); err != nil {
+		log.Fatalf("Error with creating experiment: %v", err)
```

Comment:
As you are using Fatalf here. It will restrict you to use restartPolicy: Never( you have to use this way always) else convert Fatalf to Errorf and put a return statement

Reply:
Should I keep Fatalf only then?

Reply:
Both are fine. But If you are using Fatalf then you have to make sure that restartPolicy should be Never otherwise it may create a problem for negative cases

Reply:
Ok used Errorf.